core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1747,7 +1747,7 @@ abstract class RDD[T: ClassTag](
   // =======================================================================
```

Comment:
Not related to this PR but shall we remove `@Experimental` and `@Since("3.0.0")` at `getResourceProfile`, `withResources`, etc. (probably in a separate PR)? It's counterintuitive to mark it `private` but annotate it as APIs.

Reply:
We should annotate it properly when we actually make it public. I think I pointed out somewhere but the counter argument was that it's going to be public in Spark 3.0. But seems making public is not going to happen in Spark 3.0 (?).

Reply:
cc @tgravescs

Reply:
correct, it won't be public in 3.0 as the feature didn't get completely in. I have a jira to make them public in 3.1 and will update the tags at that point. If you want to remove them for 3.0 that is fine.

Reply:
Let's fix then. I don't think it's right to keep things for future or just in case. Things ideally happen in-place.

Reply:
@tgravescs, mind fixing them please? are `withResources` and `getResourceProfile` all instances to fix?

Reply:
sure, I'll put up pr after a bit.

Reply:
Actually sorry these aren't in spark 3.0 branch, its only in master, so no reason to change as I will be doing a jira in the next week or so to make these public

Reply:
I see. Thanks for double checking @tgravescs.